### ISSUE TITLE
DLSS FrameGeneration in Portal Prelude RTX 30series

### ISSUE THREAD

**Issue by Z4ndyz** [emoose/DLSSTweaks#105]
Hello emoose

Wasn't really sure where to open this issue since is not strictly related to DlssTweaks but maybe you could take a look into it

I stumbled upon this video recently https://www.youtube.com/watch?v=k9zczxncsos and i had a look at it myself

I went up and downloaded DLSS FG DLL 1.0.1 and it did work in Portal RTX Prelude and i could use it to boost my framerate from 48-60fps up to 100-110fps range on my 3070, the setting did work on 30 series and it was kinda smooth but it's very buggy at low base framerate, i was stuttering hard at 70-80fps with FG on but that's not the point.

The point is i wanted to know if you could look up into the hooking and if you have any idea on why this works and if it could work with dlsstweaks for other more stable framegen dlls, i was using dlsstweaks for a lower dlss ultra performance ratio.

Posted some screenshots below (48fps to 101fps after toggling dlss fg using 1.0.1 dll), OSD goes darker and haywire by turning itself on/off after toggling dlss fg and if i toggle fg too many times it'll crash having me to restart my pc before i can run dlss fg back again in portal prelude rtx.

Original dll or any other dll than 1.0.1 doesn't work and it's very weird that this dll works in particular and seems to do something while also proving rtx 30 series are capable of handling FG maybe there is a hook to be bypassed.

![Screenshot (1523)](https://github.com/emoose/DLSSTweaks/assets/43518302/b5a61a43-e812-451a-877d-0b74039c7267)
![Screenshot (1525)](https://github.com/emoose/DLSSTweaks/assets/43518302/a13501ff-2c3e-43ce-b8b9-7656d6c5ba9b)
![Screenshot (1526)](https://github.com/emoose/DLSSTweaks/assets/43518302/6a98fcf5-184b-428e-8869-57c2d4e74f8c)
![Screenshot (1528)](https://github.com/emoose/DLSSTweaks/assets/43518302/25a63906-f267-43d2-bef2-9b615694a057)
![Screenshot (1529)](https://github.com/emoose/DLSSTweaks/assets/43518302/3acfdf5a-f55f-4c76-8940-039dfe26e265)

**Comment by PweSol** [comment]
Nvidia's FG is artificially locked behind the 40 series. It has happened a few times that people found a way to unlock it for all GPUs in some specific games with early FG implementation. Nvidia has apparently patched that out. Nobody has found a universal solution on how to unlock RTX features. Hopefully, someone will eventually, or Nvidia will do so for their FG now that AMD has finally released their alternative.

It would be nice if DLSSTweaks could just do that.

The closest thing that comes to my mind is swapping your VBIOS to the 40 series. But that obviously comes with severe risks.

**Comment by Z4ndyz** [comment]
Further information
It does work in Portal RTX if you copy paste the bin content and replace it with the Portal RTX Prelude, it seems to be tied to the bridge version or something else in dll too.
Tldr : Dll 1.0.1 fg + bridge version from Portal Prelude RTX works

**Comment by doktorsleepelss** [comment]
I just tried it with my 2070S.  My fps did pretty much double, but It doesn't actually feel or look like it doubled.   Also tons of screen tearing.

**Comment by jpgsantos** [comment]
I can confirm this works in my 2070 super. The frame passing was messed up at first but if I set Vsync to a little bit less than half my monitor refresh rate in the nvidia control panel everything works as it should. Approximately double the frames and acceptable latency, I feel a little bit cheated by NVIDIA right now.

**Comment by NoUserNameForYou** [comment]
It's proven by SebaRTX on Reddit that Ampere could do FG.

Even if you take the official Nvidia claim that Ada has 2.5x faster OFA (which is quite possibly an edge case where a certain algorithm is tested because on some other algos a 4070ti is 30% faster than 3090ti) a frame's generation takes multiple variables into account. Ampere could increase a 50 fps game upto 80 fps whereas ADA would upto 100.

Thank you for your discovery. I hope this blows in Nvidia's face.

**Comment by jpgsantos** [comment]
Official NVIDIA response:
"Why does it seem like DLSS Frame Generation works on RTX 20 and 30 series GPUs in Portal Prelude RTX?

A: We’ve noticed that some members of the PC gaming community, like Discord Communities before it, have encountered a bug in an old v1.0.1.0 DLSS DLL that falsely appears to generate frames on 20/30-series hardware. There is no DLSS Frame Generation occurring—just raw duplication of existing frames. Frame counters show a higher FPS, but there is no improvement in smoothness or experience. This can be verified by evaluating the duplicated frames using tools like NVIDIA ICAT, or viewing the error message in the Remix logs indicating an error with frame generation.

DLSS frame generation DLLs have since fixed this bug, and NVIDIA will be issuing an update to RTX Remix in the future to prevent further confusion when frame generation is behaving incorrectly."

https://www.nvidia.com/en-us/geforce/forums/geforce-graphics-cards/5/528723/portal-prelude-rtx-dlss-frame-generation-question/

**Comment by Kim2091** [comment]
This was discussed in the RTX Remix server (presumably the community they referred to). There has been proof posted that it is not working properly. If you join the server, you can look at it here: https://discord.com/channels/1028444667789967381/1028444668918235197/1157485308720455771

When recorded with a capture card, you can indeed see that the frames are just duplicates of previous frames. If you go frame-by-frame, you'll observe that there is no true interpolation. This was done on an RTX 3060 ti in Portal RTX, but was also tested in multiple other Remix games.

**Comment by PanosX24** [comment]
So is not working???is placebo?

**Comment by ATELow-dev** [comment]
Nope. Just tried it with 3080. Even with high base framerate like 80+ it didn't work. Just jittery duplicated frames. I think that hardware acceleration of motion flow for 40x series is what makes it easy for NVIDIA to prevent this to work properly on 30, 20 series cards.

**Comment by Panjaksli** [comment]
Did anybody try this with FSR3 mod ? I wanted to try it in normal Portal RTX, but there is no FG toggle in the menu :(
Does anyone have the old remix files from Portal Prelude to try to enable FG ?

**Comment by Z4ndyz** [comment]
View this https://github.com/Nukem9/dlssg-to-fsr3/issues/141
As of now you can get it to load but frames will be duplicated / not working properly no interpolation, it's the lack of Vulkan support (fsr 3.0 didnt launch with vulkan support), it can be modded in but the author might not have time for it rn
Need to wait until either he or smb else or AMD adds support in the dll
